### PR TITLE
Create 0000-deal-with-peer-dependencies.md

### PR DESCRIPTION
**Pull request by alloy**
A proposal to deal with peer dependency noise.

When reviewing use the instructions found in [the template](https://github.com/alloy/rfcs/blob/master/0000-template.md) as a guide.

### CODE REVIEW

File: accepted/0000-deal-with-peer-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+has a version requirement on it using a semantic-version operator such as `^`.
+E.g. `graphql@0.13.0` will not satisfy a requirement like `graphql@^0.12.0`._
+
+# Detailed design
+
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with Yarn to understand, and for somebody familiar with the
+implementation to implement. This should get into specifics and corner-cases,
+and include examples of how the feature is used. Any new terminology should be
+defined here.
```
Comment (orta) [inline accepted/0000-deal-with-peer-dependencies.md]: this is the template ^

**Review by orta — APPROVED** [review]
Yep, feels good

File: accepted/0000-deal-with-peer-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    }
+}
+```
+
+However, this breaks compatibility with npm and, more importantly, doesn’t allow
+for multiple peer dependencies of the (transitive) dependency to be ignored.
+
+# Unresolved questions
+
+I don’t feel strong about the configuration, except that I dislike more
```
Comment (ashfurrow) [inline accepted/0000-deal-with-peer-dependencies.md]: > I don’t feel strong**ly** about

File: accepted/0000-deal-with-peer-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+However, there are legitimate cases when the developer may want to ignore a peer
+dependency warning. Such as the host not being compatible with the dependency,
+the version constraint being too restrictive, or otherwise not being interested
+in the dependency.
+
+While treating warnings as errors is often useful, warnings are specifically not
+errors, because they could possibly be ignored after judgment by the developer.
+
+Currently the inability to deal with peer dependency warnings, other than
```
Comment (ashfurrow) [inline accepted/0000-deal-with-peer-dependencies.md]: > Currently **,** the inability

Missing comma.

**Review by ashfurrow — APPROVED** [review]
I like this a lot! Thorough but concise. Well-written 👏

File: accepted/0000-deal-with-peer-dependencies.md (hunk 1 above)
Comment (alloy) [reply accepted/0000-deal-with-peer-dependencies.md]: Doh! Thanks 🙈

**Comment by alloy** [comment]
Thanks @orta & @ashfurrow !

**Comment by l2succes** [comment]
Nice one @alloy!

One suggestion I have is consider mentioning there would still be the ability to output suppressed peer-dependencies warnings by using the `--verbose` flag

**Comment by anandaroop** [comment]
This is great!

**Comment by alloy** [comment]
@l2succes Done 👍

**Comment by alloy** [comment]
Going to close this and create a new upstream PR. Thanks everybody!

**Pull request closed without merging**